vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -152,9 +158,14 @@ import { extractDateParts, parseDate as _parseDate } from '@vaadin/date-picker/s
           const formatterAndParser = createFormatterAndParser(usedFormats);
 
           // Merge current web component I18N settings with new I18N settings and the formatting and parsing functions
-          // Defer setting I18N property into microtask as a workaround for: https://github.com/vaadin/flow-components/issues/4500
-          queueMicrotask(() => {
-            datepicker.i18n = Object.assign({}, datepicker.i18n, i18n, formatterAndParser);
+          const updatedI18n = Object.assign({}, datepicker.i18n, i18n, formatterAndParser);
+          datepicker.i18n = updatedI18n;
+          // Update I18N property again after the element is ready as a workaround for both:
+          // https://github.com/vaadin/flow-components/issues/4500
+          // https://github.com/vaadin/flow-components/issues/4667
+          // This workaround is only necessary for v23, and will be removed in v24.
+          datepicker.addEventListener("date-picker-ready", () => {
+            requestAnimationFrame(() => datepicker.i18n = updatedI18n);
```

Comment:
Prefer microtask timing. With animation frame, you can see the wrong locale flasing really briefly on page load.

Reply:
It actually works locally without the animation frame. But I could not get the TeamCity builds to pass. That is the reason I added that part. Microtask did not work as well. Do you have any suggestions?

Reply:
Ok, I see. Maybe we could then just add a synchronous call `datepicker.i18n = updatedI18n;` before the rAF line to avoid the flash.

Another option could be to use (instead of the custom event) a custom property change observer for `i18n` which always reverts to the desired value. [Example](https://github.com/vaadin/flow-components/blob/2d60208657c8c4353455d740cbe1bf3cad5c5e90/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js#L1002) in grid connector.

Reply:
I think this last option really makes sense. I updated the PR based on that approach.